### ISSUE TITLE
Problems with the KMT_STEER_WHEEL in some games

### ISSUE THREAD

**Issue by Bren1n-SCP** [barry-ran/QtScrcpy#353]
I have this problem in CodM where the steering wheel crashes. I think it is because the movement is very fast and therefore the movement is not registered. does anyone know how i can solve this problem? I think adding some delay to the movement, but I don't understand programming.

**Comment by renanwp2** [comment]
Some contributor will need to patch the source code for it. It looks that berry has not been here for a while. All movements should have some delay, I suppose. The best way to test the needed delay is using [xbindkeys](https://wiki.archlinux.org/index.php/Xbindkeys) for moving the mouse and doing the clicks.

**Comment by Bren1n-SCP** [comment]
well, thanks. i'm gonna study this when i have time. can you give me your tag of discord to have more contact and help together ?

**Comment by Yamakuzure** [comment]
> It looks that barry has not been here for a while. 

12 days ago he was, and the issue he created looks like he's still interested. Probably just busy.
See: #344 
Translated to english: https://translate.google.com/translate?hl=&sl=zh-CN&tl=en&u=https%3A%2F%2Fgithub.com%2Fbarry-ran%2FQtScrcpy%2Fissues%2F344

**Comment by Yamakuzure** [comment]
> I have this problem in CodM where the steering wheel crashes.

What does that mean exactly?

> I think it is because the movement is very fast and therefore the movement is not registered.

Please describe what
a. is happening and
b. what should happen instead in your opinion.

> does anyone know how i can solve this problem? I think adding some delay to the movement, but I don't understand programming.

Well, I do, and I am currently looking for ways to improve QtScrcpy to be more usable with my (only) fun FPS game on Android.
So, maybe I have already seen the same issue as you have, but wasn't able to clearly identify it. So any pointers are highly appreciated.

**Comment by Bren1n-SCP** [comment]
well, when I map the keyboard with the "kmt_steer_wheel" option in Call of Duty mobile and try to move, sometimes the game doesn't recognize what I'm pressing the button on the keyboard and it just doesn't work. as if the movement that the "kmt_steer_wheel" does is very fast, so the game does not capture the drag of the finger that is made by mapping the keyboard. And I need to press the button again to have the chance for the game to capture the finger drag that is done by the keymapper.

a.when i press the buttons in which i mapped with the QtScrcpy ("kmt_steer_wheel" option) to walk around in codm (call of duty mobile), the steer wheel movement of the game, do not register the displacement of the finger done by the keymapper.
b.I think adding the option to put custom delay on the "kmt_steer_wheel" during the displacement of the "centerpos" to the "leftoffset", "rightoffset", "upoffset" and "downoffset".

sorry for my bad english, i'm from brazil. :(

**Comment by Yamakuzure** [comment]
@Bren1n-SCP : Your description is very good!

Actually I believe that you just explained a problem I see in "FRAG! Pro Shooter", which makes it almost unplayable with QtScrcpy. What you describe can explain the issues I am seeing.

I do have a rough idea where to put such a delay in. Let's see whether we can get this done.

**Comment by Yamakuzure** [comment]
Your issue seems to be exactly the same as in #352.
If you think that I am correct, we could unify the two by referencing the other and closing this issue.

**Comment by Bren1n-SCP** [comment]
Yeah! It is the same answer as "a." question you asked him.. Thank you for all that. But how can i close this issue ? (i'm new here)

**Comment by Bren1n-SCP** [comment]
that's the way i close this issue ?

**Comment by renanwp2** [comment]
You can't comment after you have closed the issue. Any comments you have to add to the issue need to be added before closing it, if you need it.

**Comment by trivita** [comment]
yeah, i have that too, in CoDM
i tested with using only KMT_STEER_WHEEL
when you press WSAD quickly enough, it will eventually go crazy. like WSAD will be stuck or jumpy

**Comment by barry-ran** [comment]
i will test

**Comment by owel0426** [comment]
> i will test

Yes please. I have the same issue in CODM. Your work is very helpful. Emulators cost a lot of cpu resources. Looking forward a fix for this issue.

**Comment by sickofrain** [comment]
> i will test

Thanks for the great software, barry!
I'm having the same problem with another game.
I hope it will be solved.

**Comment by barry-ran** [comment]
@sickofrain What a surprise, you are my first sponsor, thank you very much for your recognition，HAHAHA. LOL :)

**Comment by barry-ran** [comment]
> > i will test
> 
> Thanks for the great software, barry!
> I'm having the same problem with another game.
> I hope it will be solved.

What game are you talking about

**Comment by sickofrain** [comment]
> @sickofrain What a surprise, you are my first sponsor, thank you very much for your recognition，HAHAHA. LOL :)

Thank you so much.
my pleasure! =)

> What game are you talking about

I'm talking about 荒野行为.

**Comment by barry-ran** [comment]
@sickofrain  Sorry to update so late, you can try [v1.7.0](https://github.com/barry-ran/QtScrcpy/releases/tag/v1.7.0)

**Comment by Bren1n-SCP** [comment]
i'm still having this delay problem playing minecraft pocket edition with the new beta mcpe controls. Have any way to custom this delay of "KMT_STEER_WHEEL" by myself ?
